riscv/zvk_ext_macros.h

```diff
@@ -86,6 +86,41 @@
 //    (LMUL * VLEN) <= EGW
 #define require_egw_fits(EGW)  require((EGW) <= (P.VU.VLEN * P.VU.vflmul))
 
+// Ensures that a register index is aligned with LMUL
+#define require_vreg_align_lmul(VREG_NUM) \
+  if (P.VU.vflmul > 1) { \
+    require_align(VREG_NUM, P.VU.vflmul); \
+  }
+
+// Ensures that a register index is aligned to EMUL
+// evaluated as EGW / VLEN.
+// The check is only enabled if this value is greater
+// than one (no index alignment check required for fractional EMUL)
+#define require_vreg_align_eglmul(EGW, VREG_NUM) \
+  do { \
+    float vfeglmul = EGW / P.VU.VLEN; \
+    if (vfeglmul > 1) { \
+      require_align(VREG_NUM, vfeglmul); \
+    }\
+  } while (0)
+
+#define require_vd_align_lmul require_vreg_align_lmul(insn.rd())
+#define require_vs2_align_lmul require_vreg_align_lmul(insn.rs2())
+#define require_vs1_align_lmul require_vreg_align_lmul(insn.rs1())
+#define require_vs2_align_eglmul(EGW) require_vreg_align_eglmul(EGW, insn.rs2())
+
+// ensure that rs2 and rd do not overlap, assuming rd encodes an LMUL wide
+// vector register group and rs2 encodes an vs2_EMUL=ceil(EGW / VLEN) vector register
+// group.
+// Assumption: LMUL >= vs2_EMUL which is enforced independently through require_egw_fits.
+#define require_no_overlap_eglmul(vd, vs2) \
+  do { \
+  int vd_emul = P.VU.vflmul < 1.f ? 1 : (int) P.VU.vflmul; \
+  int aligned_vd = vd / vd_emul; \
+  int aligned_vs2 = vs2 / vd_emul; \
+  require(aligned_vd != aligned_vs2); \
```

Comment:
Could also be implemented by masking the `log2(max(LMUL, 1))` LSBs of the register indices.
Need to check if there is not already a macro to perform this function.